corehq/apps/case_search/models.py

```diff
@@ -112,11 +118,18 @@ def _validate_multiple_terms(self):
     def _validate_daterange(self):
         if not self.is_daterange:
             return
-
         pattern = re.compile(r'__range__\d{4}-\d{2}-\d{2}__\d{4}-\d{2}-\d{2}')
-        match = pattern.match(self.value)
-        if not match:
-            raise CaseFilterError(_('Invalid date range format, {}'), self.key)
+        if self.has_multiple_terms:
+            # don't validate empty values
+            values = self._value_without_empty()
+            if isinstance(values, str):
```

Comment:
ah, I missed that `_value_without_empty()` also flattens the list.

Maybe worth just doing the filtering directly?:
```
values = [val for val in self.value if val != '']
```